task: remove hostname v1 (#32352)
keycloak/keycloak: docs/documentation/release_notes/topics/26_0_0.adoc

Reviewer: This is essentially a breaking change. Could you please also add this paragraph to upgrading guide as well? There, we could also mention/link the 25 upgrading guide where we provided detaileds steps for v1 to v2 migration.

Author: Sounds good. I didn't see a corresponding upgrading guide entry for the account console v2 theme removal, so I wasn't sure if we needed both entries.

of `localhost`. As an alternative to the `_LEGACY` cookies Keycloak now doesn't set the `secure` flag and sets `SameSite=Lax`
instead of `SameSite=None` when it detects an insecure context is used.

= Hostname v1 feature removed

> The deprecated hostname v1 feature was removed. This feature was deprecated in {project_name} 24 and replaced by the hostname v2 theme. If you are still using this feature, you should migrate to the hostname v2 feature.

Reviewer 2: ```suggestion
The deprecated `hostname:v1` feature was removed. This feature was deprecated in {project_name} 25 and replaced by the hostname v2 theme. If you are still using this feature, you must migrate to the `hostname:v2` feature.

For more details, see the https://www.keycloak.org/server/hostname[Configuring the hostname (v2)].
```
1. I think it was deprecated in [Keycloak 25](https://www.keycloak.org/2024/06/keycloak-2500-released.html#_new_hostname_options), right? 
2. Maybe formatting like this might be better? As users dealing with the `--features=hostname:vX` syntax?
3. Perhaps it would be good to explicitly mention the guide for hostname v2?

Author: > I think it was deprecated in [Keycloak 25](https://www.keycloak.org/2024/06/keycloak-2500-released.html#_new_hostname_options), right?

Yes, sorry this was a cut and paste error from the account console removal.

> Maybe formatting like this might be better? As users dealing with the --features=hostname:vX syntax?

I don't have a strong opinion on that. We on have 4 references in the docs to the hostname: format, all the other references are to hostanme v.

> Perhaps it would be good to explicitly mention the guide for hostname v2?

That's fine. Here again I was just copying the other removal notice - it seemed to assume users were aware from the deprecation notice / warnings.

Reviewer 2: > it seemed to assume users were aware from the deprecation notice / warnings.

@shawkins Our release notes contain a lot of information, and it's very easy to get lost, so I'd be more explicit in this case and mention it again to be sure. Thanks for the update!

> 

Reviewer 2: I'd explicitly mention that the `proxy` option has been finally removed, and users need to migrate to the `proxy-headers`?

Author: Yes, I'll add that too.

of `localhost`. As an alternative to the `_LEGACY` cookies Keycloak now doesn't set the `secure` flag and sets `SameSite=Lax`
instead of `SameSite=None` when it detects an insecure context is used.

= Hostname v1 feature removed

> The deprecated hostname v1 feature was removed. This feature was deprecated in {project_name} 25 and replaced by hostname v2. If you are still using this feature, you must migrate to hostname v2. For more details, see the https://www.keycloak.org/server/hostname[Configuring the hostname (v2)].

Reviewer 2: ```suggestion
The deprecated `hostname:v1` feature was removed. This feature was deprecated in {project_name} 25 and replaced by `hostname:v2`. If you are still using this feature, you must migrate to hostname v2 options. For more details, see the https://www.keycloak.org/server/hostname[Configuring the hostname (v2)].
```

I'd like to distinguish the hostname v2 **feature** and **options**. IMHO, when we are talking about features, we implicitly mean Keycloak features, and it's natural for me to use the same syntax as for specifying the Keycloak features (`--features=hostname:v2`).

@shawkins I'd prefer this if it's ok for you. WDYT?

Author: > @shawkins I'd prefer this if it's ok for you. WDYT?

It seems like a follow-up issue would be needed to establish a convention and update the other feature references. 

cc @vmuzikar

Reviewer 2: Follow-up works for me.

Reviewer: To be fair, in docs we [reference](https://www.keycloak.org/server/features#_deprecated_features) it as `hostname-v1`. 😄 Not sure that's correct but it's a follow-up in any case.

Author: Unfortunately there are the following:
- feature label / description
- enum name: name_vX
- feature key: name-vX
- option key: name:vX
- properties file key (as it's used in testing): keycloak.profile.feature.name_vX (the later part should match the enum name)
- and other places use hostname vX

= Proxy option remove

> The deprected proxy option was removed. This option was deprecated in {project_name} 24 and replaced by the proxy-headers option in combination with hostname options as needed. For more details, see https://www.keycloak.org/server/reverseproxy[using a reverse proxy] and https://www.keycloak.org/docs/latest/upgrading/index.html#deprecated-proxy-option[the previous upgrading guide].

Reviewer 2: ```suggestion
The deprecated `proxy` option was removed. This option was deprecated in {project_name} 24 and replaced by the `proxy-headers` option in combination with hostname options as needed. For more details, see https://www.keycloak.org/server/reverseproxy[using a reverse proxy] and https://www.keycloak.org/docs/latest/upgrading/index.html#deprecated-proxy-option[the previous upgrading guide].
```
